mozdownload/parser.py

```diff
@@ -13,8 +13,8 @@
 class DirectoryParser(HTMLParser):
     """Class to parse directory listings"""
 
-    def __init__(self, url, authentication=None, timeout=None):
-        self.authentication = authentication
+    def __init__(self, url, session=None, timeout=None):
+        self.session = session or requests.Session()
```

Comment:
So if we allow `None` for the session it would mean we cannot authenticate anymore and would fail, right?